src/gs-screenshot-image.c

```diff
@@ -435,12 +552,33 @@ gs_screenshot_image_load_async (GsScreenshotImage *ssimg,
 
 	ssimg->message = soup_message_new_from_uri (SOUP_METHOD_GET, base_uri);
 	if (ssimg->message == NULL) {
+		if (showing_image) {
+			g_warning ("Could not setup a new soup_message for the "
+				   "URL '%s'. Showing cached image '%s'.", url,
+				   ssimg->filename);
+			return;
+		}
+
+		/* silently return as we have already a screenshot set, even if
+		 * eventually outdated */
+		if (showing_image)
+			return;
+
 		/* TRANSLATORS: this is when networking is not available */
 		gs_screenshot_image_set_error (ssimg, _("Screenshot not available"));
 		return;
 	}
 
+	if (showing_image)
+		gs_utils_set_modified_request (ssimg->message, ssimg->filename);
+
+	if (writable_filename != NULL) {
```

Comment:
No. We do this when the writable_filename is valid _and_ exists. So at this point, the writable_filename may point to the user's cache path _but_ we're actually showing the system cache's image (and have filename assigned as such).